backend/remote-state/gcs/backend.go

```diff
@@ -162,22 +147,22 @@ func (b *Backend) configure(ctx context.Context) error {
 			return fmt.Errorf("Error parsing credentials '%s': %s", contents, err)
 		}
 
-		conf := jwt.Config{
-			Email:      account.ClientEmail,
-			PrivateKey: []byte(account.PrivateKey),
-			Scopes:     []string{storage.ScopeReadWrite},
-			TokenURL:   "https://oauth2.googleapis.com/token",
-		}
-
-		opts = append(opts, option.WithHTTPClient(conf.Client(ctx)))
-	} else {
-		opts = append(opts, option.WithScopes(storage.ScopeReadWrite))
+		credOptions = append(credOptions, option.WithCredentialsJSON([]byte(contents)))
 	}
 
 	// Service Account Impersonation
 	if v, ok := data.GetOk("impersonate_service_account"); ok {
 		ServiceAccount := v.(string)
-		opts = append(opts, option.ImpersonateCredentials(ServiceAccount))
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: ServiceAccount,
+			Scopes:          []string{storage.ScopeReadWrite},
+		}, credOptions...)
+
+		if err != nil {
+			return err
+		}
+
+		opts = append(opts, option.WithTokenSource(ts))
```

Comment:
Won't two `CredentialsConfig` options get added when there are delegates? How is that handled, and can we link to documentation if so?

Reply:
That is wrong. I have fixed it now.